core/templates/services/platform-feature.service.ts

```diff
@@ -146,20 +152,31 @@ export class PlatformFeatureService {
       if (item && this.validateSavedResults(item)) {
         PlatformFeatureService.featureStatusSummary = item.featureStatusSummary;
         this.saveResults();
-      } else {
-        this.clearSavedResults();
+        return;
       }
+      this.clearSavedResults();
 
-      if (!PlatformFeatureService.featureStatusSummary) {
-        PlatformFeatureService.featureStatusSummary = await this
-          .loadFeatureFlagsFromServer();
-        this.saveResults();
+      // The user is 'partially logged-in' at the signup page, we need to skip
+      // the loading from server otherwise the request will have the cookies
+      // erased, leading to the 'Registration session expired' error.
+      if (this.urlService.getPathname() === '/signup') {
+        PlatformFeatureService._isSkipped = true;
+        PlatformFeatureService.featureStatusSummary = this
+          .featureStatusSummaryObjectFactory.createDefault();
+        return;
```

Comment:
Will the features be correctly loaded before the user reaches a non-signup page?

Reply:
Also, is this scenario being tested?

Reply:
Thanks for clarifying!